scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -167,6 +166,19 @@ def _final_handle_actions(self, harness, handler_time):
 
         self.record(harness)
 
+    def get_default_domain_build_dir(self):
+        if self.instance.testsuite.sysbuild:
+            # Load domain yaml to get default domain build directory
+            # Note: for targets using QEMU, we assume that the target will
+            # have added any additional images to the run target manually
+            domain_path = os.path.join(self.build_dir, "domains.yaml")
+            domains = Domains.from_file(domain_path)
+            logger.debug("Loaded sysbuild domain data from %s" % domain_path)
+            build_dir = domains.get_default_domain().build_dir
```

Comment:
Thanks @gchwier . This fixes the issue @nordicjm reported.
An extra thought though: This will work well when there is only 1 image, or, for multi-image cases, when sysbuild's default image is the image in which everything got assembled together (which for the testcases we have today in tree, we are lucky that we always chose the DEFAULT_IMAGE).
I wonder if, for native targets , we should either have domains.cmake have as default whichever was set as the "final_image" in
https://github.com/zephyrproject-rtos/zephyr/blob/main/share/sysbuild/cmake/modules/native_simulator_sb_extensions.cmake#L5-L28
or something else that would cover the case in which the default != final assembled image.

Reply:
Good point @aescolar
I've added that fix just for native targets, to be able to run it with sysbuild configuration without having to duplicate sample/testcase configs. I didn't find that for `native` could be multi-image case, also sysbuild is not really used for native targets.
I didn't update `build_dir` for `bsim` target, so you can still use `native_simulator_set_final_executable` cmake method (https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/harness.py#L696)
Since I don't have a complete understanding of the multi-image and bsim I don't want to do more changes in this PR. Let's iterate on further changes when multi-image support becomes necessary (at least for bsim)